pkg/db/src/implementations/version-1.2/books.ts

```diff
@@ -52,41 +64,17 @@ class Books implements BooksInterface {
 				filter: (doc) => isbns.includes(doc._id.replace("books/", ""))
 			});
 
-			const initialState = from(
-				new Promise<PouchDB.Core.AllDocsResponse<BookEntry>>((resolve) => {
-					this.#db._pouch
-						.allDocs<BookEntry>({ keys: isbns.map((isbn) => `books/${isbn}`), include_docs: true })
-						.then((res) => {
-							debug.log(ctx, "books_stream:initial_query:result")(res);
-							return resolve(res);
-						})
-						.catch(debug.log(ctx, "books_stream:initial_query:error"));
-				})
-			);
+			const initialState = from(this.get(isbns));
 
+			// only use latest change to query allDocs
 			const changeStream = newChangesStream<BookEntry[]>(ctx, emitter).pipe(
 				// The change only triggers a new query (as changes are partial and we need the "all docs" update)
-				switchMap(() =>
-					from(
-						new Promise<PouchDB.Core.AllDocsResponse<BookEntry>>((resolve) => {
-							this.#db._pouch
-								.allDocs<BookEntry>({ keys: isbns.map((isbn) => `books/${isbn}`), include_docs: true })
-								.then((res) => {
-									debug.log(ctx, "books_stream:change_query:res")(res);
-									return resolve(res);
-								})
-								.catch(debug.log(ctx, "books_stream:change_query:error"));
-						})
-					)
-				)
+				switchMap(() => from(this.get(isbns)))
 			);
 
+			// call plugin here
```

Comment:
I don't see the point of this comment (probably left-over from previous implementation)